pkg/crc/machine/bundle/repository_test.go

```diff
@@ -110,42 +107,56 @@ func TestVersionCheck(t *testing.T) {
 	}
 
 	bundlePath := filepath.Join(dir, "crc_libvirt_4.6.1")
-	createDummyBundleContent(t, bundlePath, "0.9")
+	createDummyBundleContent(t, dir, "crc_libvirt_4.6.1", "0.9")
 	_, err = repo.Get("crc_libvirt_4.6.1.crcbundle")
 	assert.EqualError(t, err, "cannot use bundle with version 0.9, bundle version must satisfy ^1.0 constraint")
 	os.RemoveAll(bundlePath)
 
-	createDummyBundleContent(t, bundlePath, "1.1")
+	createDummyBundleContent(t, dir, "crc_libvirt_4.6.1", "1.1")
 	_, err = repo.Get("crc_libvirt_4.6.1.crcbundle")
 	assert.NoError(t, err)
 	os.RemoveAll(bundlePath)
 
-	createDummyBundleContent(t, bundlePath, "2.0")
+	createDummyBundleContent(t, dir, "crc_libvirt_4.6.1", "2.0")
 	_, err = repo.Get("crc_libvirt_4.6.1.crcbundle")
 	assert.EqualError(t, err, "cannot use bundle with version 2.0, bundle version must satisfy ^1.0 constraint")
 	os.RemoveAll(bundlePath)
 }
 
-func writeMetadata(t *testing.T, dir string, bundleInfo *CrcBundleInfo) bool {
-	metadataStr, err := json.Marshal(bundleInfo)
+func TestListBundles(t *testing.T) {
+	dir, err := ioutil.TempDir("", "repo")
 	assert.NoError(t, err)
-	return assert.NoError(t, ioutil.WriteFile(filepath.Join(dir, metadataFilename), []byte(metadataStr), 0600))
-}
+	defer os.RemoveAll(dir)
+
+	ocBinDir, err := ioutil.TempDir("", "oc-bin-dir")
+	assert.NoError(t, err)
+	defer os.RemoveAll(ocBinDir)
 
-func createDummyBundleContent(t *testing.T, bundlePath string, version string) {
-	assert.NoError(t, os.Mkdir(bundlePath, 0755))
+	createDummyBundleContent(t, dir, "crc_libvirt_4.6.15", "1.0")
+	createDummyBundleContent(t, dir, "crc_libvirt_4.7.0", "1.0")
+	createDummyBundleContent(t, dir, "crc_libvirt_4.10.0", "1.0")
```

Comment:
As part of test can we also include the custom bundles and the nightly one to make sure it also parsed by semver?
```
$ ls ~/.crc/cache/
crc_libvirt_4.7.11_1621489482  crc_libvirt_4.8.0-0.nightly-2021-05-26-021757
```

Reply:
done. crc_libvirt_4.7.11_1621489482 doesnt work with semver. It is always first :/ 

```
"crc_libvirt_4.7.11_1621489482",
"crc_libvirt_4.10.0",
"crc_libvirt_4.8.0-0.nightly-2021-05-26-021757",
"crc_libvirt_4.7.0",
"crc_libvirt_4.6.15",
``` 

It is not particularly a problem: a custom bundle can be the first element of the list.